### PR TITLE
enable vpc access connector resource

### PR DESCRIPTION
**Pull request by prateek2408**
This change enabled magic modules to generate a resource for vpc access connector. This resource would then be used in Terraform Validator.

part of {https://github.com/hashicorp/terraform-provider-google/issues/10756}

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource

```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Are there relevant tests for this new resource?

Could you add one if not?

**Comment by prateek2408** [comment]
> 

@ScottSuarez - The test needs to be added for this resource in terraform validator repo right ?

**Comment by ScottSuarez** [comment]
No no, can be added in this repo. We copy them over

Look at `mmv1/third_party/validator/tests/data`

**Comment by prateek2408** [comment]
@ScottSuarez - Thanks for the info. I am writing the test cases for this resource

File: mmv1/third_party/validator/tests/data/example_vpc_access_connector.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        "asset_type": "vpcaccess.googleapis.com/Connector",
+	"ancestry_path": "{{.Ancestry}}/project/{{.Provider.project}}",
+        "resource": {
+            "version": "v1",
+            "discovery_document_uri": "https://www.googleapis.com/discovery/v1/apis/vpcaccess/v1/rest",
+            "discovery_name": "Connector",
+	    "parent": "//cloudresourcemanager.googleapis.com/projects/{{.Provider.project}}",
+            "data": {
+                "ipCidrRange": "10.8.0.0/28",
+                "maxThroughput": 1000,
```
Comment (melinath) [inline mmv1/third_party/validator/tests/data/example_vpc_access_connector.json:13]:
```suggestion
                "maxThroughput": 300,
```

File: mmv1/third_party/validator/tests/data/example_vpc_access_connector.tfplan.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            "name": "connector",
+            "provider_name": "google",
+            "change": {
+                "actions": [
+                    "create"
+                ],
+                "before": null,
+                "after": {
+                    "ip_cidr_range": "10.8.0.0/28",
+                    "max_throughput": 1000,
```
Comment (melinath) [inline mmv1/third_party/validator/tests/data/example_vpc_access_connector.tfplan.json:41]:
```suggestion
                    "max_throughput": 300,
```

**Review by melinath — CHANGES_REQUESTED** [review]
It looks like the conversion tests are failing due to this diff:

```diff
13c13
<     (string) (len=13) "maxThroughput": (float64) 1000,
---
>     (string) (len=13) "maxThroughput": (float64) 300,
```

This seems to be because max_throughput has a [default of 300 in the Terraform provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vpc_access_connector#argument-reference), so you'd need to either update the expected value or add an explicit field to the TF config.

**Comment by prateek2408** [comment]
@melinath - Thanks for pointing this out. I have made the necessary changes to the code.

**Review by melinath — APPROVED** [review]
LGTM

**Pull request merged**